go_modules/lib/dependabot/go_modules/file_updater.rb

```diff
@@ -10,27 +10,6 @@ module GoModules
     class FileUpdater < Dependabot::FileUpdaters::Base
       require_relative "file_updater/go_mod_updater"
 
-      def initialize(dependencies:, dependency_files:, repo_contents_path: nil,
-                     credentials:, options: {})
-        super
-        return unless repo_contents_path.nil?
```

Comment:
> How do we know the caller has performed a clone? We shouldn't assume the only caller is GitHub's wrapper.

Thanks for asking about this, @thepwagner. :zap::bow: This question makes me think that there's a smaller change that we can make to achieve the desired results (i.e., removing obsolete feature flags) without losing the flexibility of working with a repo that has no local clone.

2a1d5db91bafaa1ffcea10569e43dcd9ec845b54 brings back the ability to work with a repo that has no local clone. What do you think?

Reply:
I don't dig it. I'm concerned if you run `go mod tidy` in a repository without any code, it will wipe the modules:

```
dependabot-core/go_modules/helpers on master ✗ $ find . -name "*.go" -delete && go mod tidy && GIT_PAGER=cat git diff go.mod
diff --git a/go_modules/helpers/go.mod b/go_modules/helpers/go.mod
index 96942bd2d..72e6544df 100644
--- a/go_modules/helpers/go.mod
+++ b/go_modules/helpers/go.mod
@@ -2,12 +2,6 @@ module github.com/dependabot/dependabot-core/go_modules/helpers

 go 1.13

-require (
-	github.com/Masterminds/vcs v1.13.1
-	github.com/dependabot/dependabot-core/go_modules/helpers/updater v0.0.0
-	github.com/dependabot/gomodules-extracted v1.1.0
-)
-
 replace github.com/dependabot/dependabot-core/go_modules/helpers/importresolver => ./importresolver

 replace github.com/dependabot/dependabot-core/go_modules/helpers/updater => ./updater
```

I think we either need to keep the `repo_contents_stub` guard, or enforce that `repo_contents_path` is a full checkout.

Reply:
Thanks, @thepwagner! I'm gonna make some revisions and I'll follow up soon. :bow:

Reply:
👋 @thepwagner: I've slimmed this change down a bit based on this feedback. Here's an updated summary of the changes, which is also now included in the PR body above.

- Remove the `tidy` feature flag from `Dependabot::GoModules::FileUpdater`. You'll still see the `tidy` option in `Dependabot::GoModules::FileUpdater::GoModUpdater`. That's because `Dependabot::GoModules::FileUpdater` checks creates a placeholder Git repository if we don't have a local clone of the repository.
    - [If we're using a placeholder, we don't want to perform a tidy](https://github.com/dependabot/dependabot-core/pull/2660#discussion_r508556417), so we set the `tidy` option to `false` for the `GoModUpdater`.
    - If we have a real clone of the repository, we can safely tidy, so we set the `tidy` option to `true` for the `GoModUpdater`.
- Remove the `vendor` feature flag from `Dependabot::GoModules::FileUpdater`. You'll still see the `vendor` option in `Dependabot::GoModules::FileUpdater::GoModUpdater`. That's because `Dependabot::GoModules::FileUpdater` checks for the presence of `vendor/modules.txt`.
    - If present, it sets the `vendor` option to `true` in `Dependabot::GoModules::FileUpdater::GoModUpdater`.
    - If absent, it sets the option to `false`.

How are you feeling about the overall diff at this point?

Reply:
💯 , this feels safe for historical users that aren't cloning, and enables features by default for users that are!